internal/model/datacenter.go

```diff
@@ -65,12 +65,12 @@ func (dc *Datacenter) FindHost(computeResourceReference mor) *mo.HostSystem {
 	return nil
 }
 
-// GetResourcePoolName returns the name of the Resource Pool if is not the default
-func (dc *Datacenter) GetResourcePoolName(resourcePoolReference mor) string {
+// GetResourcePool returns the name of the Resource Pool if is not the default
```

Comment:
isn't better to return an err as well when returning nil?
The function name is getResourcePool, I would not expect a nil to be returned

Reply:
This was to simplify the client and also the way I was used to program where 2 values were not allowed.
Returning an err i don't think it's the best approach. Do you think that if you cannot get the resource pool it's a real error  or there might be cases where it is expected? 

Before we were returning an empty string which was being set in the attribute of the sample. Is it "correct" to set en empty string in that attribute (resourcePoolName of the VsphereVmSample)?  Or was the empty string the equivalent to a "nil" and setting it in the attribute was an error?

I'll change it to the same approach as a map with a boolean indicating if we are returning a value or not. Seems better than an error.

Reply:
Yes exactly, I was mentioning this https://github.com/golang/go/wiki/CodeReviewComments#in-band-errors
I am totally fine with the bool approach!